lib/checkother.cpp

```diff
@@ -1505,6 +1491,26 @@ static const Token* getVariableChangedStart(const Variable* p)
     return start;
 }
 
+namespace {
+    struct CompareVariables {
+        bool operator()(const Variable* a, const Variable* b) const {
+            const int fileA = a->nameToken()->fileIndex();
+            const int fileB = b->nameToken()->fileIndex();
+            if (fileA == fileB) {
```

Comment:
hmm.. I suggest this (same code logically):

    if (fileA != fileB)
        return fileA < fileB;
    if (lineA != lineB)
        return lineA < lineB;
    return columnA < columnB;